### ISSUE TITLE
[BUG]`TypeError: Cannot read properties of undefined (reading 'settings')` at re insert table in version ^9.0.0

### ISSUE THREAD

**Issue by HiroyukiMakita** [fiduswriter/simple-datatables#375]
- [x] I have looked through the [documentation](https://fiduswriter.github.io/simple-datatables/documentation/) to try to see if this behavior is documented.
- [x] I have looked at the [demos](https://fiduswriter.github.io/simple-datatables/demos/) to see if one of them handles this, but none of them did.

**Describe the bug**
I get an error when re-displaying a table with the following steps.
1. `insert`
2. `destroy`
3. re `insert`
    error occurs

**How to reproduce**

I have created a StackBlitz here: [https://stackblitz.com/edit/daisyui-vue-vite-wxwb7r?file=package.json,src%2FApp.vue](https://stackblitz.com/edit/daisyui-vue-vite-wxwb7r?file=package.json,src%2FApp.vue) to reproduce this issue.

**Steps to reproduce the behavior**

- Run `npm install && npm run dev` in a terminal (if it does not run automatically)
- Click `UPDATE-TABLE` on the screen that appears.
The table will be displayed.
- Click `UPDATE-TABLE` again.
Now you will get an error in the console (`TypeError: Cannot read properties of undefined (reading 'settings')`).

**View errors**
Expected behavior
Each time you click `UPDATE-TABLE`, you expect to see the table redraw without error and the counter for no table change.

![image](https://github.com/fiduswriter/simple-datatables/assets/57791566/3ec23e06-aecf-4ef5-b4e2-87486a873445)

**Other Situations**
 
I have confirmed that the version `^8.0.1` installed by `npm i simple-datatable@8` does not generate any errors.

**Comment by Monkey999Human** [comment]
I get also the same error

**Comment by ray-wn** [comment]
I don't know if this workaround helps you @HiroyukiMakita  @enriqc3 @Monkey999Human?
I set the data using data:  { headings: headings, data: data }

I changed my `<table id="myTable"></table>` to `<div id="myTable"></div>`.
Now mytable.destroy() and doing mytable = new simpleDatatables.DataTable(tableElement, { myOptions }) works again.

Not sure if something isn't correctly reset or cached when using a table-element?
The ctor of the datatable contains:
```
if (dom instanceof HTMLTableElement) {
    this.dom = dom
} else {
    this.dom = document.createElement("table")
    dom.appendChild(this.dom)
}
```
So maybe because the table-element is created instead of using the existing one fixes it?
